processor/src/range/aux_trace.rs

```diff
@@ -29,102 +22,84 @@ impl AuxTraceBuilder {
-        // values that are added to the next row after the operation's execution.
+        // keep track of the last updated row in the `b_range` running product column. `b_range` is
+        // filled with result values that are added to the next row after the operation's execution.
         let mut b_range_idx = 0_usize;
-        // keep track of the next row to be included from the user op range check values.
-        let mut rc_user_op_idx = 0;
 
         // the first half of the trace only includes values from the operations.
-        for (clk, range_checks) in self.cycle_range_checks.range(0..self.values_start as u32) {
+        for (clk, range_checks) in self.cycle_lookups.range(0..self.values_start as u32) {
             let clk = *clk as usize;
 
             // if we skipped some cycles since the last update was processed, values in the last
-            // updated row should by copied over until the current cycle.
+            // updated row should be copied over until the current cycle.
             if b_range_idx < clk {
                 let last_value = b_range[b_range_idx];
                 b_range[(b_range_idx + 1)..=clk].fill(last_value);
-                q[b_range_idx..clk].fill(E::ONE);
             }
 
-            // move the column pointers to the next row.
+            // move the column pointer to the next row.
             b_range_idx = clk + 1;
 
-            // update the intermediate values in the q column.
-            q[clk] = range_checks.to_stack_value(main_trace, alphas);
-
-            // include the operation lookups in the running product.
-            b_range[b_range_idx] = b_range[clk] * inv_row_values[rc_user_op_idx];
-            rc_user_op_idx += 1;
+            b_range[b_range_idx] = b_range[clk];
+            // include the operation lookups
+            for lookup in range_checks.iter() {
+                let value = divisors.get(lookup).expect("invalid lookup value {}");
+                b_range[b_range_idx] -= *value;
+            }
         }
 
         // if we skipped some cycles since the last update was processed, values in the last
         // updated row should by copied over until the current cycle.
         if b_range_idx < self.values_start {
             let last_value = b_range[b_range_idx];
             b_range[(b_range_idx + 1)..=self.values_start].fill(last_value);
-            q[b_range_idx..self.values_start].fill(E::ONE);
         }
 
         // after the padded section of the range checker table, include `z` in the running product
         // at each step and remove lookups from user ops at any step where user ops were executed.
-        for (row_idx, (hint, lookup)) in self
-            .row_flags
+        for (row_idx, (multiplicity, lookup)) in main_trace
+            .get_column(M_COL_IDX)
             .iter()
             .zip(main_trace.get_column(V_COL_IDX).iter())
             .enumerate()
```

Comment:
nit: we should probably change "running product" to "running sum" in the comments.